File: java-adman-spring/application/src/main/java/com/example/adman/BackendConnector.java
package com.example.adman;

import com.sap.cloud.sdk.cloudplatform.connectivity.DefaultDestination;
import com.sap.cloud.sdk.cloudplatform.connectivity.DestinationAccessor;
import com.sap.cloud.sdk.cloudplatform.connectivity.HttpDestination;

public class BackendConnector {
    private static String destinationName;
    private static String url;
    private static String username;
    private static String password;
    private static String authentication;
    private static String apikey;
    private static HttpDestination destination;

    static {
        // Get the environment variables
        destinationName = System.getenv("S4_DESTINATION");
        url = System.getenv("S4_URL");
        username = System.getenv("S4_USERNAME");
        password = System.getenv("S4_PASSWORD");
        authentication = (username != null && password != null) ?
                "BasicAuthentication" : "NoAuthentication";
        apikey = System.getenv("S4_APIKEY");
        if (apikey==null) apikey="";

        if (destinationName != null){
            // Get the destination via the Destination service
            // of Cloud Foundry, in the SCP.
            destination = DestinationAccessor
                    .getDestination(destinationName).asHttp();
        }
        else {
            // Create the destination at run time,
            // using the URL and credentials
            // provided in the environment variables.
            destination = DefaultDestination.builder()
                    .property("Name", "DESTINATION")
                    .property("URL", url)
                    .property("Type", "HTTP")
                    .property("Authentication", authentication)
                    .property("User", username)
                    .property("Password", password)
                    .build().asHttp();
        }
    }

    public static HttpDestination getDestination() {
        return destination;
    }
    public static String getApikey() {
        return apikey;
    }
}

File: ts-adman/src/adman/backend-connector.ts
import { DestinationOrFetchOptions } from '@sap-cloud-sdk/connectivity';

export class BackendConnector {
    
    private static destination: DestinationOrFetchOptions;
    private static apikey: string;

    public static getDestination(): DestinationOrFetchOptions {

        if (this.destination == null) {
            // Get the environment variables
            let destinationName: string = process.env.S4_DESTINATION;
            let url: string = process.env.S4_URL;
            let username: string = process.env.S4_USERNAME;
            let password: string = process.env.S4_PASSWORD;
            
            if (destinationName != null){
                // Get the destination via the Destination service
                // of Cloud Foundry, in the SCP.
                this.destination = {
                    destinationName: destinationName
                };
            }
            else {
                // Create the destination at run time,
                // using the URI and credentials
                // provided in the environment variables.
                if(username != null && password != null) {

                    // BasicAuthentication
                    this.destination = {
                        url: url,
                        username: username,
                        password: password
                    };
                }
                else {
                    // NoAuthentication
                    this.destination = {
                        url: url
                    }
                }
            }
        }
        return this.destination;
    }

    public static getApikey(): string {
        if (this.apikey == null) {
            if (process.env.S4_APIKEY == null) 
                this.apikey = "";
            else
                this.apikey = process.env.S4_APIKEY;
        }
        return this.apikey;
    }
}
